### PR TITLE
Added support for PHP8

### PR DESCRIPTION
**Pull request by JeroenVanOort**
I would like to use this package in a project using PHP 8, so I added compatibility for it.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -27,7 +27,7 @@
     "illuminate/queue": "~5.5.0|~5.6.0|~5.7.0|~5.8.0|^6.0|^7.0|^8.0",
     "illuminate/routing": "~5.5.0|~5.6.0|~5.7.0|~5.8.0|^6.0|^7.0|^8.0",
     "illuminate/support": "~5.5.0|~5.6.0|~5.7.0|~5.8.0|^6.0|^7.0|^8.0",
-    "openzipkin/zipkin": "~2.0.2",
+    "openzipkin/zipkin": "~2.0.2|3.0.x-dev",
```
Comment (KinaneD) [inline composer.json:30]: I don't think we should support a dev release of Zipkin at this stage, once it is released we can do so.

File: tests/Zipkin/InteractsWithZipkin.php [hunk 2]
```diff
@@ -34,12 +34,11 @@ protected function createTracer(
 
     /**
      * @param  array  $spans
-     * @return array
      */
-    protected function shiftSpan(array &$spans): array
+    protected function shiftSpan(array &$spans): Span
```
Comment (KinaneD) [inline tests/Zipkin/InteractsWithZipkin.php:38]: 👍

**Review by KinaneD — COMMENTED** [review]
Hello @JeroenVanOort,
Thanks for the contribution!

I like the change to deal with spans rather than arrays. My only concern is support of the upcoming Zipkin release that is currently under development. I suggest we avoid that until v3.0 is officially released.

Other than that, it is good to go. @Mulkave anything to add?

Thanks.

File: composer.json (hunk 1 above)
Comment (JeroenVanOort) [reply composer.json:30]: I would love to avoid it too, I just hope they'll release v3 soon.

Comment (Mulkave) [reply composer.json:30]: Agreed, better keep support for stable releases only.

File: resources/docker/workspace/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,29 @@
+FROM php:8.0-cli-alpine
```
Comment (Mulkave) [inline resources/docker/workspace/Dockerfile:1]: interesting stuff! Thanks for adding this @JeroenVanOort 

Do you use it with your editor?

**Comment by Mulkave** [comment]
@JeroenVanOort  i ran `./vendor/bin/phpunit` within the `worspace` container and got the following:

```
Warning: Private methods cannot be final as they are never overridden by other classes in /var/www/html/vendor/phpunit/phpunit/src/Util/Configuration.php on line 176
PHPUnit 6.5.14 by Sebastian Bergmann and contributors.

EEE
Fatal error: Declaration of Vinelab\Tracing\Drivers\Zipkin\Propagation\IlluminateHttp::get($carrier, string $key): ?string must be compatible with Zipkin\Propagation\Getter::get($carrier, $key) in /var/www/html/src/Drivers/Zipkin/Propagation/IlluminateHttp.php on line 20
```

Are you sure that tests are passing for you?

**Comment by jcchavezs** [comment]
I think it is good to keep the 3.0.x-dev version on the PR and keep it on
hold until we release the 3.0 (which will happen very soon).

José Carlos Chávez


tor. 23. sep. 2021 kl. 15:14 skrev Abed Halawi ***@***.***>:

**Review by KinaneD — APPROVED** [review]
Will merge and release under `v2.1.0-rc` a pre-release.

**Pull request merged**